File: SortExamples/InsertionSort.js
const generateArray = require('./Global').generateArray;

function insertionSort(arr) {
    console.time('InsertionSort');
    let len = arr.length;
    let count = 0;
    for(let i=1; i<len; i++) {
        let j = i;
        let tmp = arr[i];
        while(j > 0 && arr[j-1] > tmp) {
            arr[j] = arr[j-1];
            j--;
            count++;
        }
        arr[j] = tmp;
    }
    console.log(arr);
    console.timeEnd('InsertionSort');
    return count;
}

console.log(insertionSort(generateArray(20000)));

File: SortExamples/PythonImplementation/InsertionSort.py
from Global import generateList

def insertionSort(l):
    for i in range(1, len(l)):
        j = i
        tmp = l[i]
        while l[j-1] > tmp and j - 1 >= 0:
            l[j] = l[j-1]
            j = j - 1
        l[j] = tmp
    return l

print(insertionSort(generateList(10000)))